components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -538,31 +588,37 @@ void ConversationDriver::OnSuggestedQuestionsResponse(
   DVLOG(2) << "Got questions:" << base::JoinString(suggestions_, "\n");
 }
 
-void ConversationDriver::MakeAPIRequestWithConversationHistoryUpdate(
-    mojom::ConversationTurn turn,
-    bool needs_page_content /* = false */) {
+void ConversationDriver::SubmitHumanConversationEntry(
+    mojom::ConversationTurn turn) {
+  VLOG(1) << __func__;
+  DVLOG(4) << __func__ << ": " << turn.text;
   // Decide if this entry needs to wait for one of:
   // - user to be opted-in
   // - conversation to be active
-  // - content to be retrieved
-  if (!is_conversation_active_ || !HasUserOptedIn() ||
-      (article_text_.empty() && needs_page_content)) {
-    // This function should not be presented in the UI if the user has not
-    // opted-in yet.
+  if (!is_conversation_active_ || !HasUserOptedIn()) {
+    VLOG(1) << "Adding as a pending conversation entry";
+    DCHECK(!pending_conversation_entry_)
+        << "Should not be able to add a pending conversation entry "
+        << "when there is already a pending conversation entry.";
     pending_conversation_entry_ =
         std::make_unique<mojom::ConversationTurn>(std::move(turn));
-
-    if (article_text_.empty() && needs_page_content) {
-      pending_message_needs_page_content_ = true;
+    // Pending entry is added to conversation history when asked for
+    // so notify observers.
+    for (auto& obs : observers_) {
+      obs.OnHistoryUpdate();
     }
-
-    // Invoking this before the creation of the page handler means the pending
-    // request will not be reported.
-    OnConversationEntryPending();
     return;
   }
 
   DCHECK(turn.character_type == CharacterType::HUMAN);
+  DCHECK(!is_request_in_progress_)
```

Comment:
We currently does not guarantee this, this will be hit when we
1) Open Leo from sidebar and submit a question.
2) While first request is still in progress (for example, waiting on remote responses), ask a question from location bar.

Reply:
Out of this PR's scope, but it might be better to have a request queue later.

Reply:
https://github.com/brave/brave-core/pull/21408/commits/d5a4afdfcede9bb28ddda67cea18f786ba1b1f6b LGTM for fixing DCHECK.